### ISSUE TITLE
MLFLOW_S3_ENDPOINT_URL environment variable set causes issue: Loading Artifacts Failed 

### ISSUE THREAD

**Issue by ait-dmirian** [mlflow/mlflow#9523]
### Summary

ho to reproduce issue:
simply run server with s3 as artifact destinantion
>mlflow server -h 0.0.0.0 --backend-store-uri file:///home/ubuntu/mlflow-database  --default-artifact-root s3://ait-dmirian-mlflowtest/mytests/

run
>export MLFLOW_S3_ENDPOINT_URL=s3://ait-dmirian-mlflowtest/mytests/
restart mflfow

try to view artifact via UI:
_Loading Artifacts Failed
Unable to list artifacts stored under <code>{artifactUri}</code> for the current run. Please contact your tracking server administrator to notify them of this error, which can happen when the tracking server lacks permission to list artifacts under the current run's root artifact directory._

MLFLOW logs:
botocore.exceptions.EndpointResolutionError: Custom endpoint `s3://ait-dmirian-mlflowtest/mytests/` was not a valid URI

### Notes

- Make sure to open a PR from a **non-master** branch.
- Sign off the commit using the `-s` flag when making a commit:

  ```sh
  git commit -s -m "..."
           # ^^ make sure to use this
  ```

- Include `#{issue_number}` (e.g. `#123`) in the PR description when opening a PR.

**Comment by ait-dmirian** [comment]
Yes bucket is correct however it is private so you wan’t be able to list and use it


I can make environment for you in my playground with S3 and EC2 instance, reproduce issue, and give you access

**Comment by Gekko0114** [comment]
Hi @harupy ,
Thank you so much for maintaining this amazing OSS :)
If no one is working on this issue, could you assign it to me?

**Comment by Gekko0114** [comment]
Hi @ait-dmirian,

I attempted to replicate the issue by using an EC2 instance and S3. 
However, in my env, artifacts are stored in S3 and I can view these artifacts from the MLflow UI.
The error you encountered is an `EndPointResolutionError`, so I think there might be issues with how the S3 URL is configured in your env.
Could you kindly share more information about your environment, or consider closing the issue?

**Comment by Gekko0114** [comment]
> export MLFLOW_S3_ENDPOINT_URL=s3://ait-dmirian-mlflowtest/mytests/

I missed this setting. With this setting, I could replicate the issue. I will investigate on it

**Comment by Gekko0114** [comment]
`MLFLOW_S3_ENDPOINT_URL` is used when you want to work with custom endpoints like [MinIO](https://mlflow.org/docs/latest/tracking.html?highlight=minio).

Also it seems that using `MLFLOW_S3_ENDPOINT_URL` with `default-artifact-root` is not recommended. 

> The MLflow tracking server utilizes specific reserved keywords to generate a qualified path. These environment configurations, if present in the client environment, can create path resolution issues. For example, providing --default-artifact-root $MLFLOW_S3_ENDPOINT_URL on the server side and MLFLOW_S3_ENDPOINT_URL on the client side will create a client path resolution issue for the artifact storage location. Upon resolving the artifact storage location, the MLflow client will use the value provided by --default-artifact-root and suffixes the location with the values provided in the environment variable MLFLOW_S3_ENDPOINT_URL. Depending on the value set for the environment variable MLFLOW_S3_ENDPOINT_URL, the resulting artifact storage path for this scenario would be one of the following invalid object store paths: https://<bucketname>.s3.<region>.amazonaws.com/<key>/<bucketname>/<key> or s3://<bucketname>/<key>/<bucketname>/<key>. To prevent path parsing issues, ensure that reserved environment variables are removed (unset) from client environments.

If you want to store artifacts in S3, it's sufficient to use `default-artifact-root`. 
Therefore, I think this issue is not a bug but rather a misunderstanding of how to use MLflow.

**Comment by jescalada** [comment]
I had this same error due to `MLFLOW_S3_ENDPOINT_URL` environment variable being set to my S3 URI. Unsetting it fixed the issue!

**Comment by harupy** [comment]
@ait-dmirian Feel free to reopen if using `default-artifact-root` without `MLFLOW_S3_ENDPOINT_URL ` as suggested in https://github.com/mlflow/mlflow/issues/9523#issuecomment-1762922985 by @Gekko0114.
